test/tdvp.jl

```diff
@@ -39,6 +40,80 @@ using Printf
   @test abs(inner(ψ0,ψ2)) > 0.99
 end
 
+@testset "Accuracy Test" begin
+
+  N = 4
+  tau = 0.02
+  ttotal = 1.0
+  cutoff = 1E-12
+  use_trotter = false
+
+  method = use_trotter ? "tebd" : "tdvp"
+
+  s = siteinds("S=1/2", N; conserve_qns=false)
+
+  os = OpSum()
+  for j in 1:(N - 1)
+    os += 0.5, "S+", j, "S-", j + 1
+    os += 0.5, "S-", j, "S+", j + 1
+    os += "Sz", j, "Sz", j + 1
+  end
+  H = MPO(os, s)
+  HM = prod(H)
+
+  if use_trotter
+    gates = ITensor[]
+    for j in 1:(N - 1)
+      s1 = s[j]
+      s2 = s[j + 1]
+      hj =
+        op("Sz", s1) * op("Sz", s2) +
+        1 / 2 * op("S+", s1) * op("S-", s2) +
+        1 / 2 * op("S-", s1) * op("S+", s2)
+      Gj = exp(-1.0im * tau / 2 * hj)
+      push!(gates, Gj)
+    end
+    append!(gates, reverse(gates))
+  end
+
+  Ut = exp(-im*tau*HM)
+
+  psi = productMPS(s, n -> isodd(n) ? "Up" : "Dn")
+  psix = prod(psi)
+
+  Sz_tdvp = Float64[]
+  Sz_exact = Float64[]
+
+  c = div(N,2)
+  Szc = op("Sz",s[c])
+  Nsteps = Int(ttotal / tau)
+  for step in 1:Nsteps
+    psix = noprime(Ut*psix)
+    psix /= norm(psix)
+
+    if use_trotter
+      psi = apply(gates,psi; cutoff)
+      normalize!(psi)
+    else
+      psi = tdvp(H,psi,-tau*im; cutoff, 
+                                normalize=true,
+                                exponentiate_tol=1E-12,
+                                exponentiate_maxiter=500,
+                                exponentiate_krylovdim=100)
+    end
+
+
+    push!(Sz_tdvp, expect(psi, "Sz"; site_range=c:c))
+    push!(Sz_exact, scalar(dag(prime(psix,s[c]))*Szc*psix))
```

Comment:
Looks like we should take the `real` part here since the tests are failing when the complex part is very small.